### DISCUSSION TITLE
Adding Bonacich Power Centrality

### DISCUSSION THREAD

**Discussion by hehao98** [networkx/networkx#7078] [Algorithms]
Bonacich power centrality is a classic centrality measure in social network research for characterizing the relative power of a node in different positive-sum or zero-sum exchange networks [1] [2] (10,000+ citations combined). However, it seems to be missing in `networkx`, despite being asked for multiple times in Stack Overflow ([3](https://stackoverflow.com/questions/56286708/are-there-networkx-and-or-igraph-functions-in-python-for-finding-the-alpha-and-b), [4](https://stackoverflow.com/questions/75290487/bonacich-centrality-in-python)). I believe an implementation in Python would be beneficial for many social network researchers, and I will be happy to come up with a PR implementing this. The R `igraph` package has an implementation [available](https://search.r-project.org/CRAN/refmans/igraph/html/power_centrality.html) which could be used as a reference. 

1. Bonacich, P. (1972). “Factoring and Weighting Approaches to Status Scores and Clique Identification.” Journal of Mathematical Sociology, 2, 113-120.
2. Bonacich, P. (1987). “Power and Centrality: A Family of Measures.” American Journal of Sociology, 92, 1170-1182.

**Comment by szhorvat** [comment]
At igraph we are looking for help with surveying the related literature and preparing for a robust implementation of this functionality in C. The literature survey / theoretical work could be shared with NetworkX. Please see https://github.com/igraph/rigraph/issues/903
